dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/cache/internal/ZipFileHandleImpl.java

```diff
@@ -358,7 +417,7 @@ public InputStream getInputStream(ZipFile useZipFile, ZipEntry zipEntry) throws
             }
 
             synchronized( zipEntriesLock ) {
-                zipEntries.put(entryCacheKey, entryBytes);
+            	putCacheEntry(cacheAsManifest, entryCacheKey, entryBytes);
```

Comment:
Since you have the synchronized in the putCacheEntry method now, you shouldn't need this synchronized block any longer.